src/interpreter/runtime/values/i32.js

```diff
@@ -1,11 +1,21 @@
 // @flow
+import { BaseNumber } from "./number";
 
 const bits = 32;
 const type = "i32";
 
-export function createValue(value: number): StackLocal {
+export class i32 extends BaseNumber {}
+
+export function createValueFromAST(value: number): StackLocal {
   value = (value | 0) % Math.pow(2, bits);
```

Comment:
By the way, I'm not super sure about this.

The idea is to produce an in32 (`| 0` in JS or at least let's assume so :smile: ), `Math.pow(i32)` doesn't produce an i32 according to the spec.

It should probably be written:
```js
value % Math.pow(2, bits) | 0
```